File: Algorithm Practica/Chap2-Recursion and Divide&Conquer Intro/Hanoi.cpp
   #include <stdio.h>
   void move(char getone, char putone)
	{
	    printf("%c--->%c\n", getone, putone);     //将一个盘从getone取走，放到putone
	}
	
 
    void hanoi(int n, char one, char two, char three) 
     //将n个盘从one杆，借助two杆，到达three杆子
	{
        if (n == 1) move(one, three);
        else
        {
            hanoi(n - 1, one, three, two);
            move(one, three);
            hanoi(n - 1, two, one, three);
        }
	}
 
 
	int main(void)
	{
        //汉诺塔问题
        int m;
        printf("Input the number of disks:");
        scanf("%d", &m);
        printf("The steps to moving the %d disks are as follows:\n", m);
        hanoi(m, 'A', 'B', 'C');
        system("pause");
        return 0;
	}

File: Algorithm Practica/Homework02/第二章其余算法/Hanoi.java
import java.util.Scanner;

public class Hanoi
{
    public static void main(String[] args) 
    {
        Scanner in=new Scanner(System.in);
        int n=in.nextInt();
        in.close();
        hanoi(n,'A','B','C');
    }
    public static void hanoi(int n,Character one,Character two,Character three)
    {
        if(n==1)
        {
            move(one,three);
        }
        else
        {
            hanoi(n - 1, one, three, two);
            move(one, three);
            hanoi(n - 1, two, one, three);
        }
    }
    public static void move(Character from,Character to)
    {
        System.out.printf("Move %c to %c\n",from,to);
    }
}